### DISCUSSION TITLE
What are you doing with your initial nodes?

### DISCUSSION THREAD

**Discussion by jhoelzel** [rancher/rke2#4534] [Q&A]
Hey guys,

to begin with i am in love with rke2 and its my favorite distro so far and i really appreciate all the work that is going into it =)

How are you treating the initial server of a HA environment? Since it has to be started without the server url to initialize the cluster, i have been having some thoughts about what to do with it. 

I use terraform for provisioning since it easily integrates with everything else i have and henceforth i always end up with a different machine for the intial server than for all other master. Now i can call that machine a master all day long too but the fact remains that it will always have a different config than all other master because of the server URL.

I was wondering what other people are doing with this at the point where the cluster is running.
do i:
- leave everything as is forever and mark the node in terraform as "unchangeable"?
- update the config on the initial server once the cluster is running to then include the server url too?
- provision 4 maters in total and just delete the initial one?

I am using rolling release operating systems with rke2 so im not worried about updates, but more about what is to be done once i have a hardware failure on the initial node.

Currently, when i delete it, it will provision a new cluster for itself when i "terraform over it".

Thanks in advance for your ideas

**Comment by brandond** [comment]
> do i:
> 1. leave everything as is forever and mark the node in terraform as "unchangeable"?
> 2. update the config on the initial server once the cluster is running to then include the server url too?
> 3. provision 4 maters in total and just delete the initial one?

I would probably do 1 or 2, depending on whatever's easiest for you.

**Comment by ghost** [comment]
I use HAProxy to LB kube-apiserver and rke2-api in tcp mode. I set up HAP first, then 1st server node, then others. Finally put server statement on server #1. So all servers are 'equal'. If you are fully automated, option number 3 would not require to set up this special process for node number 1 and would be simpler.
